File: code-samples/javascript-fetch/ip-detect.js
/**
 * KwikAPI SDK — IP Detect
 * ─────────────────────────────────────────────────────────────────────────────
 * Returns the public IP address of the calling server or client. No authentication required. Use this utility endpoint to discover your server's public IP for IP whitelisting — required before using the Payout API. Run this on your production server, not your local machine.
 *
 * Endpoint  : GET /api/v2/ip_detect.php
 * Group     : Utility APIs
 * Rate Limit: Unlimited
 * NOTE: No authentication required.
 * NOTE: Run on your production server for accurate IP detection.
 *
 * @param  {string} apiKey  Your KwikAPI API key
 * (no parameters)
 * @returns {Promise<object>} Parsed JSON response
 *
 * Environment:
 *   UAT (testing) : https://uat.kwikapi.com
 *   Production    : https://www.kwikapi.com  (change BASE_URL)
 *
 * @link https://kwikapi.com/auth/register  Get your API key
 */

const BASE_URL = 'https://uat.kwikapi.com'; // Switch to https://www.kwikapi.com for production

/**
 * IP Detect
 * @returns {Promise<object>}
 */
async function ipDetect() {
  try {
  const response = await fetch('https://uat.kwikapi.com/api/v2/ip_detect.php', {
    method: 'GET',
    headers: { Accept: 'application/json' },
  });

    if (!response.ok) {
      throw new Error(`HTTP error ${response.status}: ${response.statusText}`);
    }

    const data = await response.json();

    if (!data.success) {
      throw new Error(`API error: ${data.message || 'Unknown error'}`);
    }

    return data;
  } catch (error) {
    console.error('[KwikAPI] IP Detect failed:', error.message);
    throw error;
  }
}

// ── Example usage ─────────────────────────────────────────────────────────────
(async () => {
  try {
    const result = await ipDetect();
    console.log('Success:', result);
  } catch (error) {
    console.error('Failed:', error.message);
    process.exit(1);
  }
})();

File: code-samples/python/ip-detect.py
#!/usr/bin/env python3
"""
KwikAPI SDK — IP Detect
─────────────────────────────────────────────────────────────────────────────
Returns the public IP address of the calling server or client. No authentication required. Use this utility endpoint to discover your server's public IP for IP whitelisting — required before using the Payout API. Run this on your production server, not your local machine.

Endpoint  : GET /api/v2/ip_detect.php
Group     : Utility APIs
Rate Limit: Unlimited
    # NOTE: No authentication required.
    # NOTE: Run on your production server for accurate IP detection.

Requirements:
    pip install requests

Environment:
    UAT (testing) : https://uat.kwikapi.com
    Production    : https://www.kwikapi.com  (change BASE_URL)

Get your API key: https://kwikapi.com/auth/register
"""

import requests
from typing import Any

BASE_URL = "https://uat.kwikapi.com"  # Switch to https://www.kwikapi.com for production
API_KEY  = "YOUR_API_KEY"


class KwikAPIError(Exception):
    """Raised when the KwikAPI returns a non-success response."""


def ip_detect() -> dict[str, Any]:
    """
    IP Detect

    Args:
        None

    Returns:
        dict: Parsed JSON response from KwikAPI.

    Raises:
        KwikAPIError: If the API returns success=false.
        requests.HTTPError: On non-2xx HTTP status.
    """
    url = BASE_URL + "/api/v2/ip_detect.php"

    with requests.Session() as session:
        session.headers.update({"Accept": "application/json"})
        response = session.get(url)

    response.raise_for_status()
    data = response.json()

    if not data.get("success"):
        raise KwikAPIError(data.get("message", "Unknown API error"))

    return data


# ── Example usage ─────────────────────────────────────────────────────────────
if __name__ == "__main__":
    try:
        result = ip_detect()
        import json
        print(json.dumps(result, indent=2))
    except KwikAPIError as e:
        print(f"API Error: {e}")
        raise SystemExit(1)
    except requests.RequestException as e:
        print(f"Request failed: {e}")
        raise SystemExit(1)
